web/components/instance/authorization-form.tsx

```diff
@@ -0,0 +1,287 @@
+      GOOGLE_CLIENT_SECRET: config["GOOGLE_CLIENT_SECRET"],
+      GITHUB_CLIENT_ID: config["GITHUB_CLIENT_ID"],
+      GITHUB_CLIENT_SECRET: config["GITHUB_CLIENT_SECRET"],
+    },
+  });
+
+  const onSubmit = async (formData: AuthorizationFormValues) => {
+    const payload: Partial<AuthorizationFormValues> = { ...formData };
+
+    await instanceStore
+      .updateInstanceConfigurations(payload)
+      .then(() =>
+        setToastAlert({
+          title: "Success",
+          type: "success",
+          message: "Authorization Settings updated successfully",
+        })
+      )
+      .catch((err) => console.error(err));
+  };
+
+  const originURL = typeof window !== "undefined" ? window.location.origin : "";
+
+  return (
+    <div className="flex flex-col gap-8 m-8 w-4/5">
+      <div className="flex items-center gap-8 pb-4 border-b border-custom-border-100">
+        <div>
+          <div className="text-custom-text-100 font-medium text-sm">Enable sign-up</div>
+          <div className="text-custom-text-300 font-normal text-xs">
+            Keep the doors open so people can join your workspaces.
+          </div>
+        </div>
+        <div>
+          <Controller
+            control={control}
+            name="ENABLE_SIGNUP"
+            render={({ field: { value, onChange } }) => (
+              <ToggleSwitch
+                value={Boolean(parseInt(value))}
+                onChange={() => {
+                  Boolean(parseInt(value)) === true ? onChange("0") : onChange("1");
+                }}
+                size="sm"
+              />
+            )}
+          />
+        </div>
+      </div>
+
+      <div className="flex flex-col gap-y-6 py-2">
+        <Disclosure as="div">
+          {({ open }) => (
+            <div className="w-full">
+              <Disclosure.Button
+                as="button"
+                type="button"
+                className="flex items-center justify-between w-full py-2 border-b border-custom-border-100"
+              >
+                <span className="text-lg font-medium tracking-tight">Google</span>
+                {open ? <ChevronDown className="h-5 w-5" /> : <ChevronRight className="h-5 w-5" />}
+              </Disclosure.Button>
+
+              <Transition
+                show={open}
+                enter="transition duration-100 ease-out"
+                enterFrom="transform opacity-0"
+                enterTo="transform opacity-100"
+                leave="transition duration-75 ease-out"
+                leaveFrom="transform opacity-100"
+                leaveTo="transform opacity-0"
+              >
+                <Disclosure.Panel className="flex flex-col gap-8 px-2 py-8">
+                  <div className="grid grid-col grid-cols-1 lg:grid-cols-2 items-center justify-between gap-x-16 gap-y-8 w-full">
+                    <div className="flex flex-col gap-1">
+                      <h4 className="text-sm">Client ID</h4>
+                      <Controller
```

Comment:
Create a new form component and implement your form fields inside the component.